metadata-transformer/src/main/java/no/unit/nva/metadata/MetadataConverter.java

```diff
@@ -2,16 +2,106 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import no.unit.nva.api.CreatePublicationRequest;
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Pattern;
 
 import static nva.commons.core.JsonUtils.objectMapper;
 
-public final class MetadataConverter {
+public class MetadataConverter {
+
+    public static final String DOI_PREFIX = "doi:";
+    public static final String DOI_ORG = "https://doi.org/";
+    public static final String ANY_23 = "http://vocab.sindice.net/any23#";
+    public static final String DC_IDENTIFIER = "dc.identifier";
+    public static final String DC_IDENTIFIER_UPPER_CASE = "DC.identifier";
+    public static final String DOI_START = "10\\.[0-9]+\\/.*";
+    public final Pattern doiStartPattern;
+
+    private static final Logger logger = LoggerFactory.getLogger(MetadataConverter.class);
+    private static final ValueFactory valueFactory = SimpleValueFactory.getInstance();
+
+    private final Model metadata;
+    private final String jsonld;
+
+    public MetadataConverter(Model metadata, String jsonld) {
+        this.metadata = metadata;
+        this.jsonld = jsonld;
+        this.doiStartPattern = Pattern.compile(DOI_START);
+    }
 
-    private MetadataConverter() {
+    public CreatePublicationRequest toRequest() throws JsonProcessingException {
+        CreatePublicationRequest request = objectMapper.readValue(jsonld, CreatePublicationRequest.class);
+        addAdditionalFieldsFromMetadata(request, metadata);
+        return request;
     }
 
-    public static CreatePublicationRequest fromJsonLd(String jsonld) throws JsonProcessingException {
-        return objectMapper
-                .readValue(jsonld, CreatePublicationRequest.class);
+    private void addAdditionalFieldsFromMetadata(CreatePublicationRequest request, Model metadata) {
+        addDoi(request, metadata);
     }
+
+    private void addDoi(CreatePublicationRequest request, Model metadata) {
+        Optional<URI> doi = getDoiFromMetadata(metadata);
+        if (doi.isPresent()) {
+            request.addReferenceDoi(doi.get());
```

Comment:
:+1:

Reply:
This is only used here. I worry addAdditionalFieldsFromMetadata will become a shopping list method; maybe move the actual logic into that method instead of breaking it up or find another solution.